### ISSUE TITLE
Implement screenreader support for Shrine

### ISSUE THREAD

**Issue by rami-a** [material-components/material-components-flutter-gallery#56]
## Acceptance Criteria
* Study confirms to a11y standards
* Tooltips for all icon buttons
* Tabbing through layout should work on menu too

## Breakdown of Task
- [x] **11-29 (3)** Let everything be **read** correctly.
- [x] **11-29 (1)** Let everything be **reachable**.
- [x] **11-28 (1)** Do not read things that are invisible (such as a hidden menu item).
- [x] **11-28 (2)** Do not read unimportant things (such as a divider, or the Shrine logo).
- [x] **11-29 (2)** Add voice feedback to **actions** (for example, clearing the cart should announce "cart cleared and closed").

## Potential Additions
* Let every **screen** be read with a summary.
* Let every **page change** have a voice feedback (such as closing the cart).
* Fine tune: `OrdinalSortKey`
